Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp

```diff
@@ -5899,7 +5900,12 @@ namespace AzToolsFramework
 
             for (auto componentEditor : m_componentEditors)
             {
-                componentEditor->EnteredComponentMode(componentModeTypes);
+                auto enteredComponentMode = componentEditor->EnteredComponentMode(componentModeTypes);
```

Comment:
The function name is a little confusing, It checks whether the componentEditor in the for loop has entered component mode or not and if not disables them, it should be named something like ProcessComponentEditorOnEnteredComponentMode but thats horrible. Currently the false return does actually mean that component mode was not entered for that component so hasHighlightedItem doesn't make too much sense either, any suggestions?

Reply:
+1 good spot - it is also a bit confusing having the return value here be a bool as it does really look like it might be 'component mode was enabled' as @mbalfour-amzn describes - I know this came up before but maybe an enum here might actually be useful to make the code more readable? (depending on your perspective 😝)

Reply:
Resolving as I believe this has now been addressed